core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -342,6 +342,13 @@ public final double readDouble() throws IOException {
         return Double.longBitsToDouble(readLong());
     }
 
+    public final Double readOptionalDouble() throws IOException {
```

Comment:
You mean I should pull out the new tests into their own method? Makes sense to me.